common/lib/api-client/auth.js

```diff
@@ -0,0 +1,58 @@
+const axios = require('axios')
+
+function Auth({ timeout = 10000, authUrl, username, password } = {}) {
+  this.accessToken = null
+  this.tokenExpiresAt = null
+  this.config = {
+    timeout,
+    authUrl,
+    username,
+    password,
+  }
+}
+
+Auth.prototype.getCurrentTime = function() {
+  return Math.floor(new Date() / 1000)
+}
+
+Auth.prototype.getAccessToken = async function() {
+  if (this.isExpired()) {
+    return this.refreshAccessToken()
+      .then(data => this.setAccessToken(data))
+      .then(() => this.accessToken)
+  }
+
+  return Promise.resolve(this.accessToken)
+}
+
+Auth.prototype.setAccessToken = function(response) {
```

Comment:
not too sure but my spidey senses are saying this needs to resolve the `accessToken`

Reply:
What do you mean? I tried to split the concerns of `get` and `set`. This sets the token on the class, the other method returns it.

Reply:
it feels like if you also returned the access key when using set then the above (L21) could become 
```
.then(data => this.setAccessToken(data))
.then((accessToken) => accessToken)
```

But that then feels a bit like concerns are being mixed between OO and Functional

Reply:
Yeah that's true. I suppose the name didn't suggest it was returning anything but I I see your point about functional should always return something.

Could combine with the get method so it gets set there and is also always returned.